app/react/components/buttons/LoadingButton.tsx

```diff
@@ -21,19 +26,25 @@ export function LoadingButton({
       {...buttonProps}
       type={type}
       disabled={disabled || isLoading}
+      icon={LoadingButtonIcon(isLoading, icon)}
     >
-      {isLoading ? (
-        <>
-          <i
-            className="fa fa-circle-notch fa-spin space-right"
-            aria-label="loading"
-            aria-hidden="true"
-          />
-          {loadingText}
-        </>
-      ) : (
-        children
-      )}
+      {isLoading ? loadingText : children}
     </Button>
   );
 }
+
+function LoadingButtonIcon(isLoading: boolean, defaultIcon: ReactNode) {
+  if (defaultIcon && !isLoading) {
+    return defaultIcon;
+  }
+  if (isLoading) {
+    return (
+      <Icon
+        icon={CircleNotch}
+        className="animate-spin-slow ml-1"
+        aria-label="loading"
+      />
+    );
+  }
+  return null;
+}
```

Comment:
just a suggestion:
```suggestion
function loadingButtonIcon(isLoading: boolean, defaultIcon?: ReactNode) {
  if (!isLoading) {
    return defaultIcon;
  }
  
    return (
      <Icon
        icon={CircleNotch}
        className="animate-spin-slow ml-1"
        aria-label="loading"
      />
    );
  }
}
```

this is doing a fast return. if not loading just return the default icon (which can be undefined), the rest of the code happens when isLoading is true. then we return the spinner.

also notice I renamed the function name to lowerCaseCamelCase. because it's a regular and not a component it should start with lower case character

Reply:
another option is not to use a function at all. create a spinner component for the icon. and then use `icon={isLoading : <Spinner /> : icon}`

Reply:
nice suggestion for the loadingButtonIcon function - it looks a lot cleaner - updated